models/articles.js

```diff
@@ -21,35 +21,17 @@ const db = require('../db/connection');
 //     });
 // }
 
-// exports.selectArticleById = (id) =>
-// {
-//     return db.query(`SELECT *, count(article_id) FROM articles, comments WHERE article_id=${id} AND comments.article_id = article_id GROUP BY article_id;`)
-//     .then(({rows}) =>
-//     {
-//         console.log(rows);
-//         if (!rows.length)
-//         {
-//             const err = new Error(`Article ID: ${id} not found,`);
-//             err.status = 404;
-//             return Promise.reject(err);
-//         }
-//         return rows[0];
-//     })
-//     .catch((err) =>
-//     {
-//         return Promise.reject(err);
-//     });
-// };
-
-
 exports.selectArticleById = (id) =>
 {
-    return db.query(`SELECT * FROM articles WHERE article_id=${id};`)
+    return db.query(`SELECT *,
+    (SELECT count(*) FROM comments c WHERE c.article_id = a.article_id)
+    AS comment_count FROM articles a WHERE article_id = $1;`, [id])
     .then(({rows}) =>
     {
+        console.log(rows);
         if (!rows.length)
         {
-            const err = new Error(`Article ID: ${id} not found,`);
+            const err = new Error(`Not found,`);
```

Comment:
This could be simplified to one line if you send back an object {status:404, msg: "Not found"}